model/metricset.go

```diff
@@ -96,6 +106,20 @@ type Sample struct {
 	// Name holds the metric name.
 	Name string
 
+	// Type holds an optional metric type.
+	//
+	// If Type is unspecified or invalid, it will be ignored.
+	Type MetricType
+
+	// Unit holds an optional unit:
+	//
+	// - "percent" (value is in the range [0,1])
+	// - "byte"
+	// - a time unit: "nanos", "micros", "ms", "s", "m", "h", "d"
+	//
+	// If Unit is unspecified or invalid, it will be ignored.
```

Comment:
assuming this will hold true once the `Unit` is actually filled

Reply:
Unit is currently always ignored. The unit is added to the doc, but gets removed by the ingest pipeline.